File: 卡码/代码随想录/JS/二叉树/04-二叉搜索树属性/LeetCode_0700.js
//700. 二叉搜索树中的搜索
import TreeNode from "../TreeNode.js";

const searchBST = function(root, val) {
  while(root){
    if(val<root.val) root=root.left;
    else if(val>root.val) root=root.right;
    else return root;
  }
  return null;
};

File: 卡码/代码随想录/1/06-二叉树/Go/04-二叉搜索树属性/LeetCode_0700.go
//700. 二叉搜索树中的搜索
package main

import (
	. "binaryTree/types"
)

func searchBST(root *TreeNode, val int) *TreeNode {
	for root!=nil {
		if val<root.Val {
			root=root.Left
		}else if val>root.Val {
			root=root.Right
		}else{
			return root
		}
	}
	return nil
}

func main{

}
